adobe/spectrum-web-components
# [DO NO MERGE]: fix(number-field): separator interpreted as decimal

## Description

Updated the special `convertValueToNumber` case for iPhones to correctly identify user's intent of using a decimal point.
Also changed the `,` or `.` that's being replaced with the website language specific decimal character from the first occurrence left-to-right to right-to-left. (e.g. for 1,234,56 previously the value would've been 1.23456, now it is 1,234.56 to better reflect user's intent).


## Related issue(s)
- https://github.com/adobe/spectrum-web-components/issues/4531


## Motivation and context

This solves the problem of a comma used as a separator in the number-field being interpreted as a decimal point when the user didn't want that. These changes came as a necessity since iOS keyboard shows either `,` or `.` based on device's settings.

## How has this been tested?

    1. Open storybook on an iPhone
    2. Select the number-field component
    3. Enter a value that would have a separator, such as 1000
    4. Blur the number-field
    5. Remove a digit from the input's value
    6. Blur the number-field (the value should be 100)
    1. Open storybook on an iPhone
    2. Select the number-field component
    3. Enter a value that would have a separator and a decimal, such as 1000.5
    4. Blur the number-field
    5. Remove/Add a digit from/to the input's value
    6. Blur the number-field
    1. Open storybook on an iPhone
    2. Select the number-field component
    3. Enter a value that would have a separator and a decimal, such as 1000.5
    4. Blur the number-field
    5. Remove/Add a digit and a comma/dot from/to the input's value
    6. Blur the number-field

Below you can find videos of some of these testings done on two different types of keyboard.

https://github.com/adobe/spectrum-web-components/assets/67201262/ca9e1781-ff1f-437a-8139-2bfa4c4df148

https://github.com/adobe/spectrum-web-components/assets/67201262/9486c881-2e98-4284-b538-a833f2ea9ba2



## Types of changes


## Checklist




## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.
